engine/consensus/matching/engine.go

```diff
@@ -471,6 +472,14 @@ func (e *Engine) onApproval(originID flow.Identifier, approval *flow.ResultAppro
 	return nil
 }
 
+func nextUnsealedsPrintable(nextUnsealeds []*nextUnsealedResult) string {
+	nextUnsealedResultsAsJSON, err := json.Marshal(nextUnsealeds)
+	if err != nil {
+		return fmt.Sprintf("can not convert next unsealeds to json: %s", err.Error())
+	}
+	return string(nextUnsealedResultsAsJSON)
+}
+
```

Comment:
suggestion:
* we move `nextUnsealedResult` to its own file. 
* Define the type
   ```golang
    type nextUnsealedResults []*nextUnsealedResult
    ``` 
* make this a `String()` method and attach it to `nextUnsealedResults`


This would declutter `engine.go`, which is already quite long